src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php

```diff
@@ -269,8 +274,11 @@ protected function getListRepresentation(
         // generate expressions for collected parent-categories
         $parentExpressions = [];
         foreach ($idsToExpand as $idToExpand) {
+            $fieldDescriptor = $listBuilder->getFieldDescriptor('parent');
+            Assert::notNull($fieldDescriptor, 'There is no field descriptor for field "parent"');
+
```

Comment:
Looks like we could do this check just once outside the `foreach` loop.

I'm also not sure about the `Assert::notNull` here 🤔 not a fan of runtime check for cases which never really can appear (parent is hardcoded fieldDescriptor). Sure best would be `getFieldDescriptor(): FieldDescriptor;` and `findFieldDescriptor(): ?FieldDescriptor;` but at current state we can not change that behaviour would more go with:

```suggestion
        /** @var FieldDescriptorInterface $parentFieldDescriptor */
        $parentFieldDescriptor = $listBuilder->getFieldDescriptor('parent');
        foreach ($idsToExpand as $idToExpand) {
```

or `assert($parentFieldDescriptor instanceof FieldDescriptorInterface);` which is optimized by the compiler to run only on dev / test envs.

Reply:
Yeah this would totally make sense to have a version of `getFieldDescriptor` which doesn't return a nullable result.